kinesis/producer.py

```
@@ -98,7 +98,30 @@ def set_put_rate_throttle(self):
+        """Validate partition key according to AWS Kinesis requirements."""
+        if not isinstance(partition_key, str):
+            raise exceptions.ValidationError("Partition key must be a string")
+
+        if not partition_key:
+            raise exceptions.ValidationError("Partition key cannot be empty")
+
+        # Check byte length (UTF-8 encoded)
+        try:
+            encoded_key = partition_key.encode("utf-8")
+        except UnicodeEncodeError:
+            raise exceptions.ValidationError("Partition key must be valid UTF-8")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)